src/helpers/transactionPressHandler.ts

```diff
@@ -78,7 +82,22 @@ export const getMenuItems = (item: any) => {
     };
   }
 
-  let buttons = [
+  const isReadOnly =
+    store.getState().wallets.selected?.type === WalletTypes.readOnly ?? true;
+
+  const parentTxHash = ethereumUtils.getHash(item);
+  const data = swapMetadataStorage.getString(parentTxHash?.toLowerCase() ?? '');
+  const wrappedMeta = data ? JSON.parse(data) : {};
+  let parsedMeta: undefined | SwapMetadata;
+  if (wrappedMeta?.type === 'swap') {
+    parsedMeta = wrappedMeta.data as SwapMetadata;
+  }
+
+  const isRetryButtonVisible =
+    !isReadOnly && status === TransactionStatus.failed && !!parsedMeta;
```

Comment:
Could this be solved by a strongly typed storage mechanism like #4283, or some other RN database solution? @jkadamczyk and I were talking earlier about local database solutions 🤔

Reply:
@estrattonbailey yeah i like that, also solving https://github.com/rainbow-me/rainbow/pull/4214#discussion_r986309762 it seems

Reply:
I made it appear just in one place, because I managed to get rid of the logic in this file, we had duplicated handlers there, now we have only one shared by notifications and this, so there's only one place, please take a look again @derHowie and @estrattonbailey